vm/src/macros.rs

```diff
@@ -18,7 +18,8 @@ macro_rules! type_check {
         // None indicates that we have no type requirement (i.e. we accept any type)
         if let Some(expected_type) = $arg_type {
             let arg = &$args.args[$arg_count];
-            if !$crate::obj::objtype::isinstance(arg, &expected_type) {
+
+            if !$crate::obj::objtype::real_isinstance($vm, arg, &expected_type)? {
```

Comment:
I know I'm late, but... do we need it here? type checking is already one of the slowest part of code, and I'd expect most, if not all, of types used for builtin functions' typechecks to not overload `_instancecheck_`.

(same for other `real_issubclass` uses - did you check if CPython uses user's overloaded `__instancecheck__` there?)

(CPython builtin functions actually completely skip `isinstance` and directly check types with eg `PyLong_CheckExact` and `PyType_FastSubclass` macros)